server/replication.go

```diff
@@ -0,0 +1,21 @@
+package server
+
+import (
+	"encoding/binary"
+
+	"github.com/go-mysql-org/go-mysql/mysql"
+)
+
+func parseBinlogDump(data []byte) (*mysql.Position, error) {
```

Comment:
```suggestion
func parseBinlogDump(data []byte) (mysql.Position, error) {
```

in other place of this repo we directly use the struct, maybe it's small so putting it on stack.

And seems you can check the length of data to return an error